### DISCUSSION TITLE
关于 secretpad部署、隐私求交效率、联合查询效率的几个问题，请求解答，谢谢

### DISCUSSION THREAD

**Discussion by ShaneGu4Real** [secretflow/secretflow#1459] [Q&A]
您好，我有几个问题，想问下：
问题1：如果我们想实现发起多个训练任务，现在的逻辑是否是在训练的两个节点里，也就是同一个容器里启动多个任务，并行操作；我们准备部署在云环境里，假定资源是很充裕的，是否可以发起多个任务的时候，启动多个容器，进行并发操作，提高效率，而不是在单个容器里并发多个任务。
问题2：如果我们想部署secretpad all in one，因为我看教程里是直接在Linux/Mac环境里执行shell脚本把容器拉起来，可否将全家桶部署在云环境里，比如k8s里，或者还支持其他什么部署方式，部署secretpad、kuscia等全家桶，如果可以应该怎么做。
问题3：如果部署在云环境里，想要实现“单任务”双方1亿数据量级别的多字段隐私求交（差不多10个字段），假定云资源充足，是否只能在双方一共两个容器里，进行求交，还是说可以调起多个容器，并发实现一个任务，来提高效率，如果可以应该怎么做。
问题4：想问下scql效率的问题，目前可以支持双方系统，每一方单表大概多少数据量的联合查询，最大可以满足多大数据量（单表 十万？百万？千万？或者更多）的联合查询，如果达到百万级别的联合查询，需要的cpu、内存、磁盘、带宽资源大概是什么样的。
麻烦解答下疑惑，谢谢您的回复。

**Comment by zimu-yuxi** [comment]
1.你这里的容器是指kuscia容器吗？如果是这样并不支持的哈。可以考虑其他方式，比如docker部署，想要多个kuscia容器，可以多台机器做组网，部署多个kuscia容器，然后使用代理做负载均衡。或者使用k8s部署
2.allinone只是一个mvp包，并没有一键部署k8s的能力。只能secretpad和kuscia分别部署
3.不可以。你们期望的时间是多少呢？
4.请问你们场景需求多少数据量级呢？8C16G的配置，千万级数据规模查询响应在分钟级别，亿级别规模查询响应小于1h。以上是参考数据，实际情况还会受到表配置、查询语句等其他因素影响，没有百万级别的参考数据

**Reply by ShaneGu4Real** [reply]
针对回答1：我们希望在secretpad界面启动，发起多任务的时候，可以启动多个容器，进行并发计算，因为我们云资源是充裕的，但是单个容器的内存和cpu等资源是有上限的
针对回答2：如果secretpad和kuscia分开部署，可以实现center模式 和edge模式在k8s环境部署吗，如果需要部署，应该怎么部署
针对回答3：小时级别实现隐私求交，也是可以接受
针对回答4：场景需求需要千万到亿级别的数据分析
您好，麻烦看下我的回复，谢谢

**Comment by Hrainy331** [comment]
@ShaneGu4Real 您好，请问您现在解决psi的并发问题了吗
